plugins/modules/zabbix_user.py

```diff
@@ -600,15 +608,15 @@ def main():
                                                         high=True,
                                                         disaster=True)),
                                       active=dict(type='bool', default=True))),
-        timezone=dict(type='str', default='default'),
-        role_name=dict(type='str', default='User role'),
-        type=dict(type='str', default='Zabbix user', choices=['Zabbix user', 'Zabbix admin', 'Zabbix super admin']),
+        timezone=dict(type='str'),
+        role_name=dict(type='str'),
+        type=dict(type='str', choices=['Zabbix user', 'Zabbix admin', 'Zabbix super admin']),
         state=dict(type='str', default="present", choices=['present', 'absent'])
     ))
     module = AnsibleModule(
         argument_spec=argument_spec,
         required_if=[
-            ['state', 'present', ['usrgrps']]
+            ['state', 'present', ['usrgrps', 'refresh', 'rows_per_page']]
```

Comment:
Is it necessary to require these parameters with `state=present`? I think that Zabbix API itself will default them as they don't seem to be required with `user.create` API call

Reply:
The refresh and rows_per_page must be required when creating a new zabbix user.  
But, maybe those are unnecessary when updating it.  
As an alternative, I can be modified the following like. (I don't test yet, but...)

https://github.com/ansible-collections/community.zabbix/pull/382/files#diff-e484d4bad2263c86ac36f50e86e030cdfece1aae86e45755e64303286dd7a01bR440-R445

What do you think?

Reply:
For example, the role_name hadn't required to specify(we could have omitted it) because it set the default value.  
The parameter couldn't have required before in the zabbix_user module.  
But if the default value will be removed, it will be required when creating a new user.  
This is a temporary process to help that.

Reply:
By the way, the existing integration test is failed if the default values are removed only.  
I think that this will be broken backward-compatible.  
I think that we should be fixing the existing integration test isn't failed until releasing version 2.0.0.
Sorry, If I don’t correctly understand, please tell me.

Reply:
Ah, I get it.  
Sorry, that was my misunderstanding.  
If those params are None, I could create a new user by removing None from the request data with the helper_normalize_data function.  
I'll work on fixing the module based on your feedback.

Reply:
cool, thanks! :)